packages/nimble-components/src/wafer-map/modules/experimental/worker-renderer.ts

```diff
@@ -7,13 +8,51 @@ import { HoverDieOpacity } from '../../types';
 export class WorkerRenderer {
     public constructor(private readonly wafermap: WaferMap) {}
 
-    public updateSortedDiesAndDrawWafer(): void {
-        // redundant function for backwards compatibility
-        this.drawWafer();
+    public async updateSortedDiesAndDrawWafer(): Promise<void> {
+        if (this.wafermap.diesTable === undefined) {
+            return;
+        }
+        await this.setupWorker();
+        const colIndexes = Int32Array.from(
+            this.wafermap.diesTable.getChild('colIndex')!.toArray()
+        );
+        const rowIndexes = Int32Array.from(
+            this.wafermap.diesTable.getChild('rowIndex')!.toArray()
+        );
+        await this.wafermap.worker.setColIndexes(
+            transfer(colIndexes, [colIndexes.buffer])
+        );
+        await this.wafermap.worker.setRowIndexes(
+            transfer(rowIndexes, [rowIndexes.buffer])
+        );
+        await this.drawWafer();
     }
 
-    public drawWafer(): void {
-        // rendering will be implemented in a future PR
+    public async drawWafer(): Promise<void> {
```

Comment:
Had to do many changes after @munteannatan pushed his latest pr, can you please take a look if the comment still stands? Now `await this.setupWorker()` is called inside `updateSortedDies()` which is called before `drawWafer()`
![image](https://github.com/ni/nimble/assets/110180309/03249e28-8e18-4e49-ae3b-758ef611ccf4)

Reply:
(`updateSortedDies` is now `setupAndDrawWaferis`) I think I got your point, 
![image](https://github.com/ni/nimble/assets/110180309/4e764696-6a9b-4c74-8be0-c9afb64419d2)
one concern might be calling `drawWafer()` before `setupAndDrawWafer()` was called therefor not being able to draw the wafer because the worker is undefined, so far we didn't have any scenario when this happened.

Reply:
I think at a minimum, the `drawWafer` method should be doing some validation checks that it _can_ render. This could simply mean doing a null check on the `worker` field, and if it is null, then throw an Error.

Having a non-internal public method that _can_ be called out of an expected, required order of another public method is fairly non-standard. Typically, required setup work is handled in a constructor (or in `connectedCallback`), but if that is not possible for technical reasons, then I think we must come up with a more robust validation strategy (like simply throwing an error).

Reply:
I added the check 
![image](https://github.com/ni/nimble/assets/110180309/5d33cad2-85e4-42c4-a6d0-0f150baa2cb9)